### ISSUE TITLE
How to decompress the dataset?

### ISSUE THREAD

**Issue by zhimin-z** [SOM-Research/HFCommunity#12]
I tried every means to decompress the zip file, either double clicking
![image](https://github.com/SOM-Research/HFCommunity/assets/8592144/41d77cd2-313b-47e2-9991-0e68935f59da)
or `NotImplementedError: That compression method is not supported` using Python archiving library
Any means to decompress the zip file? @jlcanovas @ademait @jlcanovas

**Comment by jlcanovas** [comment]
Hi @zhimin-z!

I've just downloaded the dataset (file `dump_20231009.zip` as you show in the screenshot) and I've been able to decompress it via WinRAR/Windows11. The file seems to be correct.

If I'm not wrong, the file has been created in a Linux environment, but @ademait can confirm that.

How could we reproduce your problem?

**Comment by jlcanovas** [comment]
Hi again, 

I've also tried unzip/Ubuntu and it worked.

Cheers,

**Comment by ademait** [comment]
Hi @zhimin-z !

@jlcanovas the file was zipped in windows. But I don't think this is the source of the error.

@zhimin-z are you using Archive Utility of Mac, or any other third-party software?

Cheers

**Comment by zhimin-z** [comment]
Hi, I use both MacOS (14.0) and Ubuntu (20.04). For the MacOS, I double-click the zip file but it gives the prompt as the above. For the latter, I use **zipfile** library and [Archive](https://pypi.org/project/Archive/) respectively but both gives `NotImplementedError: That compression method is not supported` error. What can I do now?

**Comment by ademait** [comment]
Hello there @zhimin-z , 

Could you try with `pyzipper` python library? If it doesn't work we can replicate it in both MacOS and Ubuntu, but not until a few days. In that case we'll keep you updated.,

**Comment by zhimin-z** [comment]
Hi! Here is the output:
![image](https://github.com/SOM-Research/HFCommunity/assets/8592144/4e60dab8-b475-408c-a0f7-2d024034c341)
It seems to output a SQL file, but I am not sure if this is correct since `pyzipper` outputs an error every time I use it: `An error occurred: compression type 9 (deflate64)` Have you ever met with it? I am using Python 3.10.9.

**Comment by jlcanovas** [comment]
Hi @zhimin-z,

thanks for sharing your advances!

We understand that even with that error, you were able to generate the SQL, right? 

It's really a strange error, we have never found that situation. After some research, I've found that `deflate64` seems to be a [proprietary variant](https://en.wikipedia.org/wiki/Deflate#Deflate64/Enhanced_Deflate) of `deflate` in ZIP and therefore the support to such compression mechanism is quite poor in some systems (mainly open source systems, it's strange why MacOS does not support it). As a workaround, I've seen that you can use [zipfile-deflate64](https://pypi.org/project/zipfile-deflate64/) in Python.

Cheers,

**Comment by zhimin-z** [comment]
Thanks, that works! However, I attempted to convert the SQL file to some workable pandas format, but still fails it, any idea?
![image](https://github.com/SOM-Research/HFCommunity/assets/8592144/c7856e16-077d-4e5d-81fc-01d36db1c4ce)
Here is the code:
```
import pandas as pd
import sqlite3

# Step 1: Connect to the database
conn = sqlite3.connect(path_data / 'database.db')

filename = 'db.sql'

# Step 2: Execute the commands from the SQL file (if needed)
with open(path_data / filename, 'r') as sql_file:
    sql_commands = sql_file.read().split(';')
for command in sql_commands:
    if command.strip():
        conn.execute(command)

# Step 3: Query data into a DataFrame
query = "SELECT * FROM your_table_name;"
df = pd.read_sql_query(query, conn)
df.to_csv(path_data / "data.csv", index=False)

# Don't forget to close the connection
conn.close()
```
Here is the error:
```
---------------------------------------------------------------------------
OperationalError                          Traceback (most recent call last)
[/home/21zz42/L-factor/code/scrape.ipynb](https://vscode-remote+ssh-002dremote-002bdocjk-002dgpu-002d02.vscode-resource.vscode-cdn.net/home/21zz42/L-factor/code/scrape.ipynb) Cell 4 line 1
     [12](vscode-notebook-cell://ssh-remote%2Bdocjk-gpu-02/home/21zz42/L-factor/code/scrape.ipynb#X11sdnNjb2RlLXJlbW90ZQ%3D%3D?line=11) for command in sql_commands:
     [13](vscode-notebook-cell://ssh-remote%2Bdocjk-gpu-02/home/21zz42/L-factor/code/scrape.ipynb#X11sdnNjb2RlLXJlbW90ZQ%3D%3D?line=12)     if command.strip():
---> [14](vscode-notebook-cell://ssh-remote%2Bdocjk-gpu-02/home/21zz42/L-factor/code/scrape.ipynb#X11sdnNjb2RlLXJlbW90ZQ%3D%3D?line=13)         conn.execute(command)
     [16](vscode-notebook-cell://ssh-remote%2Bdocjk-gpu-02/home/21zz42/L-factor/code/scrape.ipynb#X11sdnNjb2RlLXJlbW90ZQ%3D%3D?line=15) # Step 3: Query data into a DataFrame
     [17](vscode-notebook-cell://ssh-remote%2Bdocjk-gpu-02/home/21zz42/L-factor/code/scrape.ipynb#X11sdnNjb2RlLXJlbW90ZQ%3D%3D?line=16) query = "SELECT * FROM your_table_name;"

OperationalError: near "=": syntax error
```

**Comment by jlcanovas** [comment]
Hi @zhimin-z,

that's a different issue that does not seem to be related to the initial problem. At first sight, it looks like a common Python syntax error. You should review your code. 

If you detect that your problem has to do with HFCommunity, please create a new issue. For now, I'm closing this issue, as it seems it has been solved. Thanks!

Cheers,

**Comment by ademait** [comment]
Hi there,

I'm glad the error is solved. However, I'll take a look at the compression we do so it can be used with any problems in other operative systems.

**Comment by zhimin-z** [comment]
> Hi @zhimin-z,
> 
> that's a different issue that does not seem to be related to the initial problem. At first sight, it looks like a common Python syntax error. You should review your code.
> 
> If you detect that your problem has to do with HFCommunity, please create a new issue. For now, I'm closing this issue, as it seems it has been solved. Thanks!
> 
> Cheers,

The syntax error `OperationalError: near "=": syntax error` seems to be related to the SQL file. @jlcanovas

**Comment by zhimin-z** [comment]
Would you mind sharing the SQL inquiry snippet with me so I can play with it? @ademait 
Right now, I tried different means to parse the SQL dump but none has succeeded due to the syntax error like above.

**Comment by jlcanovas** [comment]
> The syntax error `OperationalError: near "=": syntax error` seems to be related to the SQL file. @jlcanovas

Hi @zhimin-z,

please, create a different issue to address each problem. 

This issue is already closed and we would like to apply some "separation of concerns" to the project 😜

Cheers,
